wordpress-mobile/WordPress-Android
# Issue/12278 do not prompt skipped qs task right away

Fixes #12278

This PR changes the logic of picking Quick Start task that will be prompted to user.

With the changes, next Quick Start task to be prompted will be picked starting from the latest skipped task. Eg, if you skip a task, complete some other task, you will be prompted to start an uncompleted task that follows skipped task. 

Given tasks A, B, C, D, E, F:
- If you complete task C
- Get prompted to start and skip task A (we pick tasks to prompt from the top)
- Complete task B
- You will be prompted task D, which is the first incomplete task after A

If no tasks were skipped they will be prompted in normal order.

To test:
- Create brand new site and start Quick Start flow.
- Navigate to `Customize Your Site` Quick Start tasks list.
- Select `Upload Site Icon` task and complete it.
- When prompted to start `Update your site title` task, select `NOT NOW` in snackbar.
- Navigate to `Customize Your Site` Quick Start tasks list again.
- Pick a different task that is not `Set your site title`, eg. `Customize your site`.
- Complete the task and navigate back to My Site screen.
- Confirm that you are prompted to start `Create a new page` task and not `Update your site title`, that you skipped before.
- Navigate to `Grow Your Audience` Quick Start tasks list.
- Pick the last task in the list - `Explore Plans`.
- Complete the task and navigate back to My Site screen.
- Confirm that you are prompted to start `Create a new page` task and not `Update your site title`, that you skipped before.
- Complete all the Quick Start tasks apart from the `Update your site title`.
- Make sure that after you complete the second to last task, you are finally prompted to start `Update your site title` task.


PR submission checklist:
